geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegion.java

```diff
@@ -557,6 +557,34 @@ public boolean virtualPut(EntryEventImpl event, boolean ifNew, boolean ifOld,
     }
   }
 
+  @Override
+  void cmnClearRegion(RegionEventImpl regionEvent, boolean cacheWrite, boolean useRVV) {
+    if (getBucketAdvisor().isPrimary()) {
```

Comment:
This is opposite of what the comment says

Reply:
Yes, I have fixed.